clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -117,13 +128,21 @@ public MetadataRequestData data() {
     public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         Errors error = Errors.forException(e);
         MetadataResponseData responseData = new MetadataResponseData();
-        if (topics() != null) {
-            for (String topic : topics())
+        if (data.topics() != null) {
+            for (MetadataRequestTopic topic : data.topics()) {
+                String topicName;
+                // If null, set to the empty string, since the response does not allow null.
+                if (topic.name() == null)
+                    topicName = "";
+                else
+                    topicName = topic.name();
                 responseData.topics().add(new MetadataResponseData.MetadataResponseTopic()
-                    .setName(topic)
-                    .setErrorCode(error.code())
-                    .setIsInternal(false)
-                    .setPartitions(Collections.emptyList()));
+                        .setName(topicName)
+                        .setTopicId(topic.topicId())
+                        .setErrorCode(error.code())
+                        .setIsInternal(false)
+                        .setPartitions(Collections.emptyList()));
```

Comment:
Is there a reason to change the indent?

Reply:
No. My IDE did that. I will revert that.